alexandria/core/filters.py

```diff
@@ -168,45 +169,89 @@ class FileFilterSet(FilterSet):
     document_metainfo = JSONValueFilter(field_name="document__metainfo")
     active_group = ActiveGroupFilter()
     files = BaseCSVFilter(field_name="pk", lookup_expr="in")
+    only_newest = BooleanFilter(method="filter_only_newest")
+
+    def filter_only_newest(self, qs, name, value):
+        if value:
+            return qs.exclude(
+                Exists(
+                    # Find all newer versions of a file. A newer version is a file:
+                    #  - of the same variant
+                    #  - with a different ID (obviously)
+                    #  - a newer created_at timestamp
+                    #  - AND the same document
+                    models.File.objects.all()
+                    .filter(
+                        document=OuterRef("document_id"),
+                        created_at__gt=OuterRef("created_at"),
+                        variant=OuterRef("variant"),
+                    )
+                    .exclude(pk=OuterRef("pk")),
+                )
+            )
+        return qs
 
     class Meta:
         model = models.File
-        fields = ["original", "renderings", "variant", "metainfo", "files"]
+        fields = [
+            "original",
+            "renderings",
+            "variant",
+            "metainfo",
+            "files",
+            "only_newest",
+        ]
 
 
-class FileSearchFilterSet(FileFilterSet):
+class SearchFilterSet(FileFilterSet):
     query = CharFilter(method="search_files")
 
+    # Cut-off value, we don't want all the garbage matches
+    min_rank = NumberFilter(field_name="search_rank", lookup_expr="gte")
```

Comment:
This does not seem to be used anywhere. Should maybe also have a default value?

Reply:
It was an idea to have a (configurable) cut-off in the frontend. I guess it's likely not going to be used tho

Reply:
If its not being used, then it can be removed right

Reply:
You are absolutely right. Removed (incl tests)